apps/src/StudioApp.js

```diff
@@ -2116,10 +2116,31 @@ StudioApp.prototype.handleEditCode_ = function (config) {
   this.dropletTooltipManager.registerDropletBlockModeHandlers(this.editor);
 
   this.editor.on('palettetoggledone', function (e) {
-    // Reposition callouts after block/text toggle (in case they need to move)
-    $('.cdo-qtips').qtip('reposition', null, false);
+    $(window).trigger('droplet_change', ['togglepalette']);
   });
 
+  this.editor.on('selectpalette', function (e) {
+    $(window).trigger('droplet_change', ['selectpalette']);
+  });
+
+  $('.droplet-palette-scroller').on('scroll', function (e) {
+    $(window).trigger('droplet_change', ['scrollpalette']);
+  });
+
+  $(window).on('prepareforcallout', function (e, options) {
+    // qtip_config's codeStudio options block is available in options
+    if (options.dropletPaletteCategory) {
+      this.editor.changePaletteGroup(options.dropletPaletteCategory);
+      var scrollContainer = $('.droplet-palette-scroller');
+      var scrollTo = $(options.selector);
+      if (scrollTo.length > 0) {
+        scrollContainer.scrollTop(scrollTo.offset().top - scrollContainer.offset().top +
+            scrollContainer.scrollTop());
+      }
+    }
+
+  }.bind(this));
```

Comment:
note: could use ES6 arrow funcs and avoid the need to bind this, i.e.
```
$(window).on('prepareforcallout', (e, options) => {
   ...
  this.foo();
});
```
could also replace `var`'s with `const`'s